### PR TITLE
Optimize static long assignment a bit

### PR DESCRIPTION
**Pull request by colinleroy**
I've worked a bit on optimizing the basics about longs: assignments.
It gives something a little larger without -O (because I've simplified that part in codegen.c to ease matching the pattern in coptlong.c), but I think it's worth it as the results with -O are quite good ?

Before/after without -O
image

Before/after with -O
image

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -66,7 +66,7 @@ ifndef BUILD_ID
 endif
 $(info BUILD_ID: $(BUILD_ID))
 
-CFLAGS += -MMD -MP -O3 -I common \
+CFLAGS += -MMD -MP -O0 -g -I common \
```
Comment (mrdudz) [inline src/Makefile:69]: This change should probably not be included in this PR? :)

File: src/cc65.vcxproj [hunk 2]
```diff
@@ -214,4 +216,4 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
\ No newline at end of file
+</Project>
```
Comment (mrdudz) [inline src/cc65.vcxproj:219]: This change should probably not be included in this PR? :)

Comment (colinleroy) [reply src/cc65.vcxproj:219]: I'm sorry, this one is just a "no newline at end of file" change. I have no idea how to configure my editors to not add them :sweat_smile:

**Comment by mrdudz** [comment]
thanks!

**Comment by mrdudz** [comment]
@colinleroy somehow merging this PR broke my floating point branch... the float-ternary.c program breaks. I tracked it down to OptLongAssign - when i disable that one, it works again. It would be super nice if you could have a look. I tried to produce a test program that reproduces it with just longs, but no luck so far :(

**Comment by mrdudz** [comment]
working code (without optimizations)
```
; fp1 = (fp2 == 2.5f) ? 1.5f : 0.5f;
	ldy     #$03
	jsr     ldeaxysp
	jsr     pusheax

	lda     #$40
	sta     sreg+1
	lda     #$20
	sta     sreg
	lda     #$00
	ldx     #$00

	jsr     ftoseqeax
	jeq     L0004

	lda     #$3F
	sta     sreg+1
	lda     #$C0
	sta     sreg
	lda     #$00
	ldx     #$00
	jmp     L0005

L0004:
	lda     #$3F
	sta     sreg+1
	lda     #$00
	sta     sreg
	lda     #$00
	ldx     #$00

L0005:
	sta     _fp1
	stx     _fp1+1
	ldy     sreg
	sty     _fp1+2
	ldy     sreg+1
	sty     _fp1+3
```

this was optimized into

```
; fp1 = (fp2 == 2.5f) ? 1.5f : 0.5f;
;
	jsr     ldeax0sp
	jsr     pusheax

	lda     #$40
	sta     sreg+1
	lda     #$20
	sta     sreg
	lda     #$00
	tax

	jsr     ftoseqeax
	beq     L0004

	lda     #$00
	jmp     L0007

L0004:	
        lda     #$3F
	sta     _fp1+3
	lda     #$00
	sta     _fp1+2
L0007:
	sta     _fp1
	sta     _fp1+1
```

somehow the second constant value disappeared

**Comment by mrdudz** [comment]
@colinleroy PS: please also add to the comments in coptlong.c what the result of that optimization is - its a bit hard to follow right now :)

**Comment by mrdudz** [comment]
Here is a reproducer (although not proper test)
```
void test(void)
{

	asm("jeq     %g", L0004);

	asm("lda     #$3F");
	asm("sta     sreg+1");
	asm("lda     #$C0");
	asm("sta     sreg");
	asm("lda     #$00");
	asm("ldx     #$00");
	asm("jmp     %g", L0005);

L0004:
	asm("lda     #$3F");
	asm("sta     sreg+1");
	asm("lda     #$00");
	asm("sta     sreg");
	asm("lda     #$00");
	asm("ldx     #$00");

L0005:
	asm("sta     _fp1");
	asm("stx     _fp1+1");
	asm("ldy     sreg");
	asm("sty     _fp1+2");
	asm("ldy     sreg+1");
	asm("sty     _fp1+3");

}
```

Correct code is produced without OptLongAssign, with it breaks the same way as shown above

**Comment by colinleroy** [comment]
Oops, sorry again. I'll fix it!
